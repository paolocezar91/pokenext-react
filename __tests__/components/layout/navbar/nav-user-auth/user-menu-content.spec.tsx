import { render, screen, fireEvent } from "@testing-library/react";
import UserMenuContent, {
  Session,
} from "@/components/layout/navbar/nav-user-auth/user-menu-content";
import "@testing-library/jest-dom";

const baseProps = {
  pathname: "/settings",
  onSignIn: jest.fn(),
  onSignOut: jest.fn(),
};

jest.mock("next-intl", () => ({
  useTranslations: () => (key: string) => key,
  useLocale: () => "en",
}));

describe("UserMenuContent", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders sign-in and settings when no session", () => {
    render(<UserMenuContent {...baseProps} session={undefined as any} />);
    expect(screen.getByText(/menu.settings/i)).toBeInTheDocument();
    expect(screen.getByText(/menu.about/i)).toBeInTheDocument();
    expect(screen.getByText(/menu.signIn/i)).toBeInTheDocument();
    expect(screen.queryByText(/menu.signOut/i)).not.toBeInTheDocument();
  });

  it("calls onSignIn when sign-in button is clicked", () => {
    render(<UserMenuContent {...baseProps} session={undefined as any} />);
    fireEvent.click(screen.getByText(/menu.signIn/i));
    expect(baseProps.onSignIn).toHaveBeenCalled();
  });

  it("matches snapshot when no session", () => {
    const { asFragment } = render(
      <UserMenuContent {...baseProps} session={undefined as any} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders user info, sign-out, and settings when session exists", () => {
    const session: Session = {
      user: { name: "Ash", email: "ash@poke.com" } as any,
      expires: "future",
    };
    render(<UserMenuContent {...baseProps} session={session} />);
    expect(screen.getByText("Ash")).toBeInTheDocument();
    expect(screen.getByText(/menu.settings/i)).toBeInTheDocument();
    expect(screen.getByText(/menu.about/i)).toBeInTheDocument();
    expect(screen.getByText(/menu.signOut/i)).toBeInTheDocument();
    expect(screen.queryByText(/menu.signIn/i)).not.toBeInTheDocument();
  });

  it("calls onSignOut when sign-out button is clicked", () => {
    const session: Session = {
      user: { name: "Ash", email: "ash@poke.com" } as any,
      expires: "future",
    };
    render(<UserMenuContent {...baseProps} session={session} />);
    fireEvent.click(screen.getByText(/menu.signOut/i));
    expect(baseProps.onSignOut).toHaveBeenCalled();
  });

  it("shows user email if name is missing", () => {
    const session: Session = {
      user: { email: "misty@poke.com" } as any,
      expires: "future",
    };
    render(<UserMenuContent {...baseProps} session={session} />);
    expect(screen.getByText("misty@poke.com")).toBeInTheDocument();
  });

  it("matches snapshot when session exists", () => {
    const session: Session = {
      user: { name: "Brock", email: "brock@poke.com" } as any,
      expires: "future",
    };
    const { asFragment } = render(
      <UserMenuContent {...baseProps} session={session} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
