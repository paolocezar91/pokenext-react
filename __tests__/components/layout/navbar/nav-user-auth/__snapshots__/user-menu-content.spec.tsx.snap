// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenuContent matches snapshot when no session 1`] = `
<DocumentFragment>
  <ul
    class="w-full"
  >
    <li
      class="flex justify-between items-center mb-2 border-b-2 border-white border-solid px-2 py-4"
    >
      <span
        class="text-xs"
      >
        <span
          class="ml-2 text-sm text-white mb-2 font-bold"
        >
          Guest
        </span>
      </span>
    </li>
    <li
      class="h-10 m-1"
    >
      <a
        class="
    flex
    px-2
    py-2
    text-sm
    rounded
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark) bg-(--pokedex-red-darker) text-white border-transparent flex justify-between w-full"
        href="/en/settings"
      >
        <span>
          menu.settings
        </span>
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.078 2.25c-.917 0-1.699.663-1.85 1.567L9.05 4.889c-.02.12-.115.26-.297.348a7.493 7.493 0 0 0-.986.57c-.166.115-.334.126-.45.083L6.3 5.508a1.875 1.875 0 0 0-2.282.819l-.922 1.597a1.875 1.875 0 0 0 .432 2.385l.84.692c.095.078.17.229.154.43a7.598 7.598 0 0 0 0 1.139c.015.2-.059.352-.153.43l-.841.692a1.875 1.875 0 0 0-.432 2.385l.922 1.597a1.875 1.875 0 0 0 2.282.818l1.019-.382c.115-.043.283-.031.45.082.312.214.641.405.985.57.182.088.277.228.297.35l.178 1.071c.151.904.933 1.567 1.85 1.567h1.844c.916 0 1.699-.663 1.85-1.567l.178-1.072c.02-.12.114-.26.297-.349.344-.165.673-.356.985-.57.167-.114.335-.125.45-.082l1.02.382a1.875 1.875 0 0 0 2.28-.819l.923-1.597a1.875 1.875 0 0 0-.432-2.385l-.84-.692c-.095-.078-.17-.229-.154-.43a7.614 7.614 0 0 0 0-1.139c-.016-.2.059-.352.153-.43l.84-.692c.708-.582.891-1.59.433-2.385l-.922-1.597a1.875 1.875 0 0 0-2.282-.818l-1.02.382c-.114.043-.282.031-.449-.083a7.49 7.49 0 0 0-.985-.57c-.183-.087-.277-.227-.297-.348l-.179-1.072a1.875 1.875 0 0 0-1.85-1.567h-1.843ZM12 15.75a3.75 3.75 0 1 0 0-7.5 3.75 3.75 0 0 0 0 7.5Z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </li>
    <li
      class="h-10 m-1"
    >
      <a
        class="
    flex
    px-2
    py-2
    text-sm
    rounded
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark) bg-transparent flex justify-between w-full"
        href="/en/about"
      >
        <span>
          menu.about
        </span>
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12Zm11.378-3.917c-.89-.777-2.366-.777-3.255 0a.75.75 0 0 1-.988-1.129c1.454-1.272 3.776-1.272 5.23 0 1.513 1.324 1.513 3.518 0 4.842a3.75 3.75 0 0 1-.837.552c-.676.328-1.028.774-1.028 1.152v.75a.75.75 0 0 1-1.5 0v-.75c0-1.279 1.06-2.107 1.875-2.502.182-.088.351-.199.503-.331.83-.727.83-1.857 0-2.584ZM12 18a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </li>
    <li
      class="h-10 m-1"
    >
      <button
        class="
    flex
    px-2
    py-2
    text-sm
    rounded
    cursor-pointer
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark)
   bg-transparent flex justify-between w-full btn btn-primary"
        type="button"
      >
        menu.signIn
        <svg
          aria-hidden="true"
          class="octicon octicon-mark-github v-align-middle"
          data-view-component="true"
          fill="#fff"
          height="22"
          version="1.1"
          viewBox="0 0 24 24"
          width="22"
        >
          <path
            d="M12 1C5.923 1 1 5.923 1 12c0 4.867 3.149 8.979 7.521 10.436.55.096.756-.233.756-.522 0-.262-.013-1.128-.013-2.049-2.764.509-3.479-.674-3.699-1.292-.124-.317-.66-1.293-1.127-1.554-.385-.207-.936-.715-.014-.729.866-.014 1.485.797 1.691 1.128.99 1.663 2.571 1.196 3.204.907.096-.715.385-1.196.701-1.471-2.448-.275-5.005-1.224-5.005-5.432 0-1.196.426-2.186 1.128-2.956-.111-.275-.496-1.402.11-2.915 0 0 .921-.288 3.024 1.128a10.193 10.193 0 0 1 2.75-.371c.936 0 1.871.123 2.75.371 2.104-1.43 3.025-1.128 3.025-1.128.605 1.513.221 2.64.111 2.915.701.77 1.127 1.747 1.127 2.956 0 4.222-2.571 5.157-5.019 5.432.399.344.743 1.004.743 2.035 0 1.471-.014 2.654-.014 3.025 0 .289.206.632.756.522C19.851 20.979 23 16.854 23 12c0-6.077-4.922-11-11-11Z"
          />
        </svg>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`UserMenuContent matches snapshot when session exists 1`] = `
<DocumentFragment>
  <ul
    class="w-full"
  >
    <li
      class="flex justify-between items-center mb-2 border-b-2 border-white border-solid px-2 py-4"
    >
      <span
        class="text-xs"
      >
        <span
          class="ml-2 text-sm text-white mb-2 font-bold"
        >
          Brock
        </span>
      </span>
    </li>
    <li
      class="h-10 m-1"
    >
      <a
        class="
    flex
    px-2
    py-2
    text-sm
    rounded
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark) bg-(--pokedex-red-darker) text-white border-transparent flex justify-between w-full"
        href="/en/settings"
      >
        <span>
          menu.settings
        </span>
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M11.078 2.25c-.917 0-1.699.663-1.85 1.567L9.05 4.889c-.02.12-.115.26-.297.348a7.493 7.493 0 0 0-.986.57c-.166.115-.334.126-.45.083L6.3 5.508a1.875 1.875 0 0 0-2.282.819l-.922 1.597a1.875 1.875 0 0 0 .432 2.385l.84.692c.095.078.17.229.154.43a7.598 7.598 0 0 0 0 1.139c.015.2-.059.352-.153.43l-.841.692a1.875 1.875 0 0 0-.432 2.385l.922 1.597a1.875 1.875 0 0 0 2.282.818l1.019-.382c.115-.043.283-.031.45.082.312.214.641.405.985.57.182.088.277.228.297.35l.178 1.071c.151.904.933 1.567 1.85 1.567h1.844c.916 0 1.699-.663 1.85-1.567l.178-1.072c.02-.12.114-.26.297-.349.344-.165.673-.356.985-.57.167-.114.335-.125.45-.082l1.02.382a1.875 1.875 0 0 0 2.28-.819l.923-1.597a1.875 1.875 0 0 0-.432-2.385l-.84-.692c-.095-.078-.17-.229-.154-.43a7.614 7.614 0 0 0 0-1.139c-.016-.2.059-.352.153-.43l.84-.692c.708-.582.891-1.59.433-2.385l-.922-1.597a1.875 1.875 0 0 0-2.282-.818l-1.02.382c-.114.043-.282.031-.449-.083a7.49 7.49 0 0 0-.985-.57c-.183-.087-.277-.227-.297-.348l-.179-1.072a1.875 1.875 0 0 0-1.85-1.567h-1.843ZM12 15.75a3.75 3.75 0 1 0 0-7.5 3.75 3.75 0 0 0 0 7.5Z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </li>
    <li
      class="h-10 m-1"
    >
      <a
        class="
    flex
    px-2
    py-2
    text-sm
    rounded
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark) bg-transparent flex justify-between w-full"
        href="/en/about"
      >
        <span>
          menu.about
        </span>
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12Zm11.378-3.917c-.89-.777-2.366-.777-3.255 0a.75.75 0 0 1-.988-1.129c1.454-1.272 3.776-1.272 5.23 0 1.513 1.324 1.513 3.518 0 4.842a3.75 3.75 0 0 1-.837.552c-.676.328-1.028.774-1.028 1.152v.75a.75.75 0 0 1-1.5 0v-.75c0-1.279 1.06-2.107 1.875-2.502.182-.088.351-.199.503-.331.83-.727.83-1.857 0-2.584ZM12 18a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
    </li>
    <li
      class="h-10 m-1"
    >
      <button
        class="
    flex
    px-2
    py-2
    text-sm
    rounded
    cursor-pointer
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark)
   bg-transparent flex justify-between w-full btn btn-primary"
        type="button"
      >
        <span>
          menu.signOut
        </span>
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 24 24"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.5 3.75A1.5 1.5 0 0 0 6 5.25v13.5a1.5 1.5 0 0 0 1.5 1.5h6a1.5 1.5 0 0 0 1.5-1.5V15a.75.75 0 0 1 1.5 0v3.75a3 3 0 0 1-3 3h-6a3 3 0 0 1-3-3V5.25a3 3 0 0 1 3-3h6a3 3 0 0 1 3 3V9A.75.75 0 0 1 15 9V5.25a1.5 1.5 0 0 0-1.5-1.5h-6Zm5.03 4.72a.75.75 0 0 1 0 1.06l-1.72 1.72h10.94a.75.75 0 0 1 0 1.5H10.81l1.72 1.72a.75.75 0 1 1-1.06 1.06l-3-3a.75.75 0 0 1 0-1.06l3-3a.75.75 0 0 1 1.06 0Z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
